### DISCUSSION TITLE
Is there a flag or setting to summarize additional steps required at the end of an install?

### DISCUSSION THREAD

**Discussion by NerdyDeedsLLC** [Homebrew/discussions#4094] [Everyday usage]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

**Is there a flag or setting to summarize additional steps required at the end of an install?**

I've dug through the docs twice now, and I don't THINK there is, but before I go writing something that will accomplish this I wanted to ask the experts.

Basically, I have a series of setup scripts intended to be run by end-users at our org. Homebrew factors heavily in certain of these. There are a number of binaries that, after Pouring, will say something like, `"Add XXX to your PATH variable,"` or "`Yo, we didn't symlink YYY,"` etc. As you know, these messages occur, inline, within the wall of output that brew occasionally produces. 

Is there a mechanism to collate (or duplicate) all of those "special case" messages into a summary block at the _end_ of the install?

Yes, I entirely agree the users should just scroll up and RTFS. But there is no amount of nagging, pointing this out, asserting, stating, opining, bullyragging, reminding, demanding, begging, or threatening that seems to be adequate to make this happen, at least at my org.

It'll be a trivial matter to just read the output, `grep` out anything that doesn't start with `Fetching`, `Downloading` , `Installing`, `Pouring`, `Regenerating`, `Running`, or `🍺  /` (I'm sure I'm missing an edge case, but off the cuff that's what occurs), and slap the output on-screen at the conclusion of the run, but it dawns on me I'm very likely not the first person to deal with this, so I figured I'd ask if there was something inbuilt and I just wasn't seeing it.

**Comment by p-linnane** [comment]
These are `caveat` blocks that are implemented on a per formula/cask level. They are printed after each formula/cask is installed, so there's no built-in mechanism to only print them after everything finished.

**Reply by gromgit** [reply]
Actually, Homebrew _does_ reprint multiple caveat blocks in a single section at the end of an install. so that fits @NerdyDeedsLLC's requirements nicely:
```
$ brew install abuse aescrypt-packetizer ahoy aircrack-ng
[...]
==> Installing abuse
==> Pouring abuse--0.8_1.arm64_monterey.bottle.tar.gz
==> Caveats
Game settings and saves will be written to the ~/.abuse folder.
==> Summary
🍺  /opt/homebrew/Cellar/abuse/0.8_1: 556 files, 22MB
==> Running `brew cleanup abuse`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Pouring aescrypt-packetizer--3.16.arm64_monterey.bottle.tar.gz
==> Caveats
To avoid conflicting with our other AESCrypt package the binaries
have been renamed paescrypt and paescrypt_keygen.
==> Summary
🍺  /opt/homebrew/Cellar/aescrypt-packetizer/3.16: 6 files, 150.3KB
==> Running `brew cleanup aescrypt-packetizer`...
==> Pouring ahoy--2.0.2.arm64_monterey.bottle.tar.gz
==> Caveats
===== UPGRADING FROM 1.x TO 2.x =====

If you are upgrading from ahoy 1.x, note that you'll
need to upgrade your ahoyapi settings in your .ahoy.yml
files to 'v2' instead of 'v1'.

See other changes at:

https://github.com/ahoy-cli/ahoy

==> Summary
🍺  /opt/homebrew/Cellar/ahoy/2.0.2: 5 files, 3.4MB
==> Running `brew cleanup ahoy`...
==> Pouring aircrack-ng--1.7_1.arm64_monterey.bottle.tar.gz
==> Caveats
Run `airodump-ng-oui-update` install or update the Airodump-ng OUI file.
==> Summary
🍺  /opt/homebrew/Cellar/aircrack-ng/1.7_1: 108 files, 3.0MB
==> Running `brew cleanup aircrack-ng`...
==> Caveats
==> abuse
Game settings and saves will be written to the ~/.abuse folder.
==> aescrypt-packetizer
To avoid conflicting with our other AESCrypt package the binaries
have been renamed paescrypt and paescrypt_keygen.
==> ahoy
===== UPGRADING FROM 1.x TO 2.x =====

If you are upgrading from ahoy 1.x, note that you'll
need to upgrade your ahoyapi settings in your .ahoy.yml
files to 'v2' instead of 'v1'.

See other changes at:

https://github.com/ahoy-cli/ahoy

==> aircrack-ng
Run `airodump-ng-oui-update` install or update the Airodump-ng OUI file.
```

**Reply by p-linnane** [reply]
TIL...thank you very much for the correction @gromgit!

**Reply by gromgit** [reply]
To be fair, that final caveat summary isn't set apart very well; the only visual difference between the summary header and the individual caveat headers...is the color of the prefix arrow.
